app/MMB/src/scenes/explore/Explore.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import * as React from 'react';
+import { graphql, PrivateApiRenderer } from '@kiwicom/mobile-relay';
+
+import BookingDetailContext from '../../context/BookingDetailContext';
+import ExploreBookingOneWay from './ExploreBookingOneWay';
+import ExploreBookingMulticity from './ExploreBookingMulticity';
+import type { ExploreQuery as ExploreQueryType } from './__generated__/ExploreQuery.graphql';
+
+function getValidExploreType(data) {
+  if (data) {
+    switch (data.__typename) {
+      case 'BookingOneWay':
+        return <ExploreBookingOneWay data={data} />;
+      case 'BookingMulticity':
+        return <ExploreBookingMulticity data={data} />;
+      default:
+        return null;
+    }
+  }
+  return null;
```

Comment:
This looks like a component. Please move it a separate file and have it fetch __typename in a fragment.